### PR TITLE
Discord Token Deleter

### LINKED ISSUE #24: Migration of pull request #159 from MorpheusHelper repo
The following pull request was created before the migration to version 3 of the MorpheusHelper. The result of this is that the pull request must be migrated to current version too.

PyDrocsid/MorpheusHelper#159

### PR DESCRIPTION
**Pull request by Tert0**
**Description**
Deletes Discord Bot Tokens in messages

**Issue**
Closes #24

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: general/discord_bot_token_deleter/cog.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+from PyDrocsid.cog import Cog
+from ...contributor import Contributor
+from PyDrocsid.translations import t
+from PyDrocsid.logger import get_logger
+from discord import Message, Embed, Forbidden, NotFound, HTTPException
+from PyDrocsid.material_colors import MaterialColors
+import re
```
Comment (Defelo) [inline general/discord_bot_token_deleter/cog.py]: Please use PyCharm to reorder the imports

File: general/discord_bot_token_deleter/translations/en.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+discord_bot_token_deleter_title: Discord Bot Token Deleter
+discord_bot_token_deleter_description: >
```
Comment (Defelo) [inline general/discord_bot_token_deleter/translations/en.yml]: You don't need the `discord_bot_token_deleter_` prefix as these translations are already in the `discord_bot_token_deleter` namespace

File: general/discord_bot_token_deleter/cog.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        if not self.RE_DC_TOKEN.findall(message.content):
+            return
+        embed = Embed(title=t.discord_bot_token_deleter_title, colour=MaterialColors.bluegrey)
+        embed.description = t.discord_bot_token_deleter_description
+        embed.set_footer(text=tg.requested_by(message.author, message.author.id), icon_url=message.author.avatar_url)
+        await message.channel.send(embed=embed)
+        try:
+            await message.delete()
+        except (Forbidden, NotFound, HTTPException) as error:
+            logger.error(error)
```
Comment (Defelo) [inline general/discord_bot_token_deleter/cog.py]: Send a custom message into the alert channel by using the `send_alert` PubSub channel instead

File: general/discord_bot_token_deleter/cog.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        """
+        deletes a message if it contains a discord bot token
+        """
+        if message.author.id == self.bot.user.id:
+            return
+        if not self.RE_DC_TOKEN.findall(message.content):
+            return
+        embed = Embed(title=t.discord_bot_token_deleter_title, colour=MaterialColors.bluegrey)
+        embed.description = t.discord_bot_token_deleter_description
+        embed.set_footer(text=tg.requested_by(message.author, message.author.id), icon_url=message.author.avatar_url)
```
Comment (Defelo) [inline general/discord_bot_token_deleter/cog.py]: Mention the original author in the message content instead

File: general/__init__.py [hunk 5]
```diff
@@ -7,3 +7,4 @@
 from .remind_me import RemindMeCog
 from .utils import UtilsCog
 from .voice_channel import VoiceChannelCog
+from .discord_bot_token_deleter import DiscordBotTokenDeleter
```
Comment (Defelo) [inline general/__init__.py]: Please reorder imports

**Review by Defelo — CHANGES_REQUESTED** [review]
Please add some newlines to improve code readability.

**Review by ghost — APPROVED** [review]

**Review by Defelo — APPROVED** [review]

**Pull request merged**